### PR TITLE
Curve editor documentation (Code comments)

### PR DESCRIPTION
**Pull request by toledoal**
Hello! 😄 

This PR has code documentation for the curve editor.  Has comments on code, and documentation on props and state.

### CODE REVIEW

**Review by RaananW — COMMENTED** [review]
I will leave the decision with @msDestiny14

File: inspector/src/components/actionTabs/tabs/propertyGrids/animations/editorControls.tsx [hunk 1]
```diff
@@ -85,10 +117,14 @@ export class EditorControls extends React.Component<
         });
     };
 
+    // Recount animations
```
Comment (msDestiny14) [inline inspector/src/components/actionTabs/tabs/propertyGrids/animations/editorControls.tsx:120]: Inconsistent commenting. comments? Seems like you're going back and forth between /* */ and // not sure why.

File: inspector/src/components/actionTabs/tabs/propertyGrids/animations/timeline.tsx [hunk 2]
```diff
@@ -184,8 +219,8 @@ export class Timeline extends React.Component<
     };
 
     /**
-    * Handles the change of number of frames available in the timeline.
-    */
+     * Handles the change of number of frames available in the timeline.
+     */
```
Comment (msDestiny14) [inline inspector/src/components/actionTabs/tabs/propertyGrids/animations/timeline.tsx:223]: Missing param

File: inspector/src/components/actionTabs/tabs/propertyGrids/animations/animationListTree.tsx [hunk 3]
```diff
@@ -184,18 +212,38 @@ export class AnimationListTree extends React.Component<
         const selectAnimation = () => this.props.selectAnimation(animation);
         const toggle = () => this.toggleProperty(i);
         return (
-            <li className={this.props.selected && this.props.selected.name === animation.name ? "property sub active" : "property sub"} key={i}>
-                <div className={`animation-arrow ${this.state.animationList && this.state.animationList[i].open ? "" : "flip"}`} onClick={toggle}></div>
+            <li
+                className={
+                    this.props.selected && this.props.selected.name === animation.name
+                        ? "property sub active"
```
Comment (msDestiny14) [inline inspector/src/components/actionTabs/tabs/propertyGrids/animations/animationListTree.tsx:218]: Spacing may be a little overdone here. Just makes it hard to read with all the separated lines.

**Review by msDestiny14 — COMMENTED** [review]
I like the amount of comments (though some places the comments are for things that are self explanatory) I added a few comments just for consistency with formatting. I also like that you made long lines into separate ones. 

My final thoughts: I think you should go and give it one more small pass and be sure to document anything you think is "strange" or not self explanatory. Besides that, this is much better. 👍 

Any additional thoughts from @RaananW ?

**Comment by RaananW** [comment]
It is kind'a hard finding specific functions now, but I would want a few line comments in complex functions (or long functions that are hard to read). As the actual pull request was merged, i guess @toledoal is the best person to ask whether or not he line-commented on those functions. Other than that - much better, I will merge it the minute @msDestiny14 gives me the OK.

**Comment by toledoal** [comment]
Alright. I'll make the updates based on your comments. Thank you! :)

File: inspector/src/components/actionTabs/tabs/propertyGrids/animations/editorControls.tsx (hunk 1 above)
Comment (toledoal) [reply inspector/src/components/actionTabs/tabs/propertyGrids/animations/editorControls.tsx:120]: The single-line comments were aimed just as context comments. But I update those to match the /**/ format. np.

File: inspector/src/components/actionTabs/tabs/propertyGrids/animations/timeline.tsx (hunk 2 above)
Comment (toledoal) [reply inspector/src/components/actionTabs/tabs/propertyGrids/animations/timeline.tsx:223]: Good catch. I'll update this and look for other places this is happening

File: inspector/src/components/actionTabs/tabs/propertyGrids/animations/animationListTree.tsx (hunk 3 above)
Comment (toledoal) [reply inspector/src/components/actionTabs/tabs/propertyGrids/animations/animationListTree.tsx]: Correct. I added a new spacing :)

**Comment by deltakosh** [comment]
Ok now that 4.2 is out I can merge the file

Do you mind fixing the conflicts?

**Comment by deltakosh** [comment]
up @toledoal  :)

**Comment by toledoal** [comment]
Sure, I'll fix the comments :) asap

**Comment by toledoal** [comment]
Ready! 👍

**Comment by toledoal** [comment]
What's your recommendation about the What's new fail?

**Comment by deltakosh** [comment]
ignore this one ;)

**Pull request merged**